metsrw/premis/event.py

```diff
@@ -0,0 +1,267 @@
+from collections import OrderedDict
+from copy import deepcopy
+from lxml import etree
+
+from .. import exceptions
+from .. import utils
+
+
+class Event(object):
+    """
+    An object representing a PREMIS event element.
+
+    :param str identifier_type: Identifier type.
+    :param str identifier_value: Identifier value.
+    :param str event_type: Event type.
+    :param str event_datetime: Event date/time.
+    :param str detail: Event details.
+    :param list outcomes: List of dicts containing outcome data.
+    :param list linking_agent_identifiers: List of dicts containing linking agent identifier data.
+    :param list linking_object_identifiers: List of dicts containing linking object identifier data.
+    :raises exceptions.ParseError: If the root tag is not event.
+    """
+    def __init__(self, identifier_type, identifier_value, event_type, event_datetime, detail=None, outcomes=None, linking_agent_identifiers=None, linking_object_identifiers=None):
+        if identifier_type is None:
+            raise exceptions.ConstructError('identify_type argument is required.')
+
+        if identifier_value is None:
+            raise exceptions.ConstructError('identify_value argument is required.')
+
+        if event_type is None:
+            raise exceptions.ConstructError('event_type argument is required.')
+
+        if event_datetime is None:
+            raise exceptions.ConstructError('event_datetime argument is required.')
+
+        self.identifier_type = identifier_type
+        self.identifier_value = identifier_value
+        self.event_type = event_type
+        self.event_datetime = event_datetime
+        self.detail = detail
+        self.outcomes = outcomes
+        self.linking_agent_identifiers = linking_agent_identifiers
+        self.linking_object_identifiers = linking_object_identifiers
+
+    @classmethod
+    def parse(cls, root):
+        """
+        Create a new Event by parsing root.
+
+        :param root: Element or ElementTree to be parsed into an object.
+        :raises exceptions.ParseError: If the root is not event.
+        """
+        if root.tag != utils.lxmlns('premis') + 'event':
+            raise exceptions.ParseError('Object can only parse event elements with PREMIS namespace.')
+
+        identifier_el = root.find("premis:eventIdentifier", namespaces=utils.NAMESPACES)
+
+        if identifier_el is not None:
+            identifier_type = identifier_el.findtext("premis:eventIdentifierType", namespaces=utils.NAMESPACES)
+            identifier_value = identifier_el.findtext("premis:eventIdentifierValue", namespaces=utils.NAMESPACES)
+        else:
+            identifier_type = None
+            identifier_value = None
```

Comment:
If these are mandatory in the constructor, why allow them to be set to None here?